### PR TITLE
ENH: Add parameters for covariance and PCA controlling batched vs. non-batched route

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

Dependent on PRs being merged:
https://github.com/uxlfoundation/oneDAL/pull/3230
https://github.com/uxlfoundation/scikit-learn-intelex/pull/2561

And requiring modifications after https://github.com/uxlfoundation/scikit-learn-intelex/pull/2561

This PR adds the newly introduced parameters for covariance and PCA that control whether to use the batched or the non-batched code routes, alongside with small tests to ensure that both produce the same results.

It currently resets the parameters after each test in a non-very-robust manner, which should be modified once there is a better way.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/decomposition/tests/test_pca.py [hunk 1]
```diff
@@ -81,3 +81,30 @@ def test_sklearnex_import(hyperparameters, dataframe, queue, macro_block, grain_
...
+    rng = np.random.default_rng(seed=123)
+    S = rng.standard_normal(size=(6, 5))
+    S = S.T @ S
+    mu = rng.standard_normal(size=5)
+    X = rng.multivariate_normal(mu, S, size=20)
+
+    hyperparameters.cpu_max_cols_batched = np.iinfo(np.int32).max
+    res_batched = PCA().fit(X).components_
+
+    hyperparameters.cpu_max_cols_batched = 1
```
Comment (Vika-F) [inline sklearnex/decomposition/tests/test_pca.py:107]: I think it should be also `hyperparameters.cpu_macro_block = 5` (or other value < 20) to really test the blocked code path vs non blocked; otherwise there is only 1 block.

Comment (david-cortes-intel) [reply sklearnex/decomposition/tests/test_pca.py:107]: The batched method is used by default in most cases, so it will be triggered in the rest of the tests from sklearnex, and in the conformance tests from sklearn.

**Review by Vika-F — APPROVED** [review]
LGTM
